test/plugin/secretless_test_harness/Makefile

```diff
@@ -0,0 +1,8 @@
+build_example:
+	go build -buildmode=plugin -o example.so ./example_plugin.go
+run_echo:
+	go run ../echo/main.go
+run_proxy:
+	go run main.go
+client:
+	nc 0.0.0.0 6175
```

Comment:
We should keep away from makefiles - they tend to complicate things _really_ quickly

Reply:
@sgnn7 i'm curious, how so ? in this case it's convenient to use them for showing the value prop of the harness.

Reply:
It's convenient here but it's an additional tool that we probably don't need is the way I consider it. Makefiles also tend to get [unmaintainable](https://github.com/torvalds/linux/blob/master/Makefile) at a much quicker rate than pure Bash which we already use everywhere. Why not just have a few (or even just 1) bin script(s)?

If you're looking for a build system more complex than Bash, we should look into Meson instead since upstream tooling for many linux projects is switching away from Makefiles as they end up unmaintainable.

Reply:
I think I agree with Srdjan though for a slightly different reason: it adds one more thing a dev working on our codebase is expected to know.  In the past, at this org, we have been _really cavalier_ about the cost of the these the kinds of decisions.  We are still not nearly principled enough about them.  A single dev should _never_ be able to introduce such a dependency without a visible, public discussion.

Imo if this can be rewritten easily enough in bash it should be, even if this is cleaner.  That is, "worse solution using team tech" > "elegant solution using new tech" (as a default -- obviously at some point the scales can tip).

Just as anecdata: I've written plenty of `Makefile`s in my time, but probably not in the last few years.  So I would be googling to understand this.  In this case, it wouldn't take long at all ofc, but that's not really the point.  There will be devs on this project who have _never_ used make.

Reply:
I agree in the general sense. `make` was definitely useful to get e2e quick. Maybe for primetime we'd want a more well-though out tool or just scripts. Any-who, this part is now taken out of the PR so that's a conversation for another time.